frontend/server/src/Controllers/Problem.php

```diff
@@ -5633,4 +5634,43 @@ public static function getLibinteractiveGenForSmarty(\OmegaUp\Request $r): array
             \OmegaUp\FileHandler::deleteDirRecursively($dirname);
         }
     }
+
+    /**
+     *
+     * @return array{smartyProperties: array{payload: ProblemCollectionDetails, title: \OmegaUp\TranslationString}, entrypoint: string}
+     *
+     * @omegaup-request-param string $collection_type
+     */
+    public static function getCollectionsDetailsForSmarty(\OmegaUp\Request $r): array {
+        \OmegaUp\Validators::validateStringNonEmpty(
+            $r['collection_type'],
+            'collection_type'
+        );
+
+        $collectionProblems = null;
+        $problemsByLevel = false;
+
+        if ($r->ensureString('collection_type') !== 'author') {
+            $problemsByLevel = true;
```

Comment:
Si esto sólo lo usas para ver que título vas a mostrar, puedes sustituir por:
```suggestion
            $title = new \OmegaUp\TranslationString(
                'omegaupTitleCollectionsByLevel'
            );
```
Y en el `else` puedes agregar algo como:
```
            $title = new \OmegaUp\TranslationString(
                'omegaupTitleCollectionsByAuthor'
            );
``` 
de esa forma, en el `return` ya puedes usar directamente la variable y evitar el operador ternario, algo como:
```
        return [
            'smartyProperties' => [
                'payload' => [
                    'collectionProblems' => $collectionProblems,
                ],
                'title' => $title,
            ],
            'entrypoint' => 'problem_collections_details',
        ];
```
Y ya después puedes eliminar L5651